fhir-swagger-generator/src/main/java/com/ibm/watson/health/fhir/openapi/generator/FHIROpenApiGenerator.java

```diff
@@ -828,13 +817,27 @@ private static void generateProperty(StructureDefinition structureDefinition, Cl
             many = true;
         }
 
-        Class<?> fieldClass = (Class<?>) fieldType;
-        ElementDefinition elementDefinition = getElementDefinition(structureDefinition, modelClass, fieldName,
-                fieldClass);
-        String description = null;
-        if (elementDefinition != null) {
-            description = elementDefinition.getDefinition().getValue();
+        String elementName = ModelSupport.getElementName(field);
+
+        if (ModelSupport.isChoiceElement(modelClass, elementName)) {
+            Set<Class<?>> choiceElementTypes = ModelSupport.getChoiceElementTypes(modelClass, elementName);
+            ElementDefinition elementDefinition = getElementDefinition(structureDefinition, modelClass, elementName + "[x]");
+            String description = elementDefinition.getDefinition().getValue();
+            for (Class<?> choiceType : choiceElementTypes) {
```

Comment:
I found the following result in the generated openapi document, and seems we combine the field name with the type names to create multiple fields, not sure if we can use "oneof" in this case... 

                           "subjectCodeableConcept": {
                                "$ref": "#/components/schemas/CodeableConcept",
                                "description": "A code or group definition that describes the intended subject of the activity being defined."
                            },
                            "subjectReference": {
                                "$ref": "#/components/schemas/Reference",
                                "description": "A code or group definition that describes the intended subject of the activity being defined."
                            },

Reply:
Combining the field name with the typeName actually produces the proper element name for choice elements  (i.e. if the type is Reference we want a serialized field name of "subjectReference" not "subject").

I considered using "oneOf", but to me it looks like that can only be used at the top level, meaning you'd need to repeat the full list of properties for each permutation of choice elements.  That just isn't feasible when there are multiple choice elements in the same resource type.

Reply:
I also verified that this representation of choice elements is consistent with the JSON Schema distributed with the specification.  They only us "oneOf" in places where a field can be any resource (e.g. Bundle.entry.resource).